docs/source/reference-lowlevel.rst

```diff
@@ -56,6 +56,35 @@ Global statistics
 .. autoclass:: RunStatistics()
 
 
+.. _trio_contexts:
+
+The current Trio context
+------------------------
+
+There are two different types of contexts in :mod:`trio`. Here are the
+semantics presented as a handy table. Choose the right function for
+your needs.
+
++---------------------------------+-----------------------------------+------------------------------------+
+| situation                       | :func:`trio.lowlevel.in_trio_run` | :func:`trio.lowlevel.in_trio_task` |
++=================================+===================================+====================================+
+| inside a running async function | `True`                            | `True`                             |
++---------------------------------+-----------------------------------+------------------------------------+
+| without a running Trio loop     | `False`                           | `False`                            |
++---------------------------------+-----------------------------------+------------------------------------+
+| in a guest run's host loop      | `True`                            | `False`                            |
++---------------------------------+-----------------------------------+------------------------------------+
+| inside an instrument call       | depends                           | depends                            |
```

Comment:
Can we make `after_run` execute before clearing `runner` so that we can give a consistent "yes" answer to whether instruments execute `in_trio_run`?